pkg/controller/shared_state.go

```diff
@@ -1277,7 +1279,7 @@ func (c *SharedState) createAzVolumeFromPv(ctx context.Context, pv v1.Persistent
 	if pv.Spec.NodeAffinity != nil && pv.Spec.NodeAffinity.Required != nil {
 		desiredAzVolume.Status.Detail.AccessibleTopology = azureutils.GetTopologyFromNodeSelector(*pv.Spec.NodeAffinity.Required, c.topologyKey)
 	}
-	if azureutils.IsMultiNodePersistentVolume(pv) {
+	if azureutils.IsMultiNodePersistentVolume(pv) || !c.mountReplicasEnabled {
```

Comment:
This check would be redundant to https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/02ecd983aaf68aaa0184f5b04b574e2ad17bb830/pkg/controller/shared_state.go#L1321

Reply:
fixed